torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -565,6 +634,15 @@ struct GraphFuser {
           return std::make_pair(consumer->reverseIterator(), true);
         }
         if(shouldFuse(consumer, producer)) {
+          // Insert explicit aten::expand nodes to inputs of simple mappable
+          // operators to make the inputs the same size, if necessary.
+          if (isSimpleMap(consumer) && !isFusable(consumer)) {
+            insertExplicitExpands(consumer);
+          }
+          // if producer->node() is not fusable, it must be simple mappable.
+          if (!isFusable(producer->node())) {
+            insertExplicitExpands(producer->node());
```

Comment:
It might be a better idea to put the expand-producing logic into `fuse` rather than trying to make a distinction in `isFusible`. Really the first part here is just trying to establish whether we can make a mutation, and the `fuse` has traditionally done the mutate part. It would be better to keep it that way and teach `fuse` how to generate expands.